VideoLocker/src/test/java/org/edx/mobile/view/CourseUnitNavigationActivityTest.java

```diff
@@ -367,4 +375,73 @@ public void orientationChangeTest() {
                 activity.getResources().getConfiguration().orientation);
         assertOrientationSetup(activity);
     }
+
+    /**
+     * Testing creation of various fragments in the {@link CourseUnitNavigationActivity}'s
+     * ViewPager, by supplying its {@link CourseUnitPagerAdapter} with all possible
+     * {@link CourseComponent} models.
+     */
+    @Test
+    public void testUnitFragmentCreation() {
```

Comment:
Could we run an iterative tests like this example?
https://github.com/edx/edx-app-android/blob/master/VideoLocker/src/test/java/org/edx/mobile/test/ConfigTests.java#L105

In this test, if any assertion fails, we will not know if any failures beyond that.